src/sentry/utils/redis.py

```diff
@@ -53,27 +57,76 @@ def make_rb_cluster(*args, **kwargs):
     return _make_rb_cluster(*args, **kwargs)
 
 
+class _RBCluster(object):
+    def supports(self, config):
+        return not config.get('is_redis_cluster', False)
+
+    def factory(self, **config):
+        # rb expects a dict of { host, port } dicts where the key is the host
+        # ID. Coerce the configuration into the correct format if necessary.
+        hosts = config['hosts']
+        hosts = {k: v for k, v in enumerate(hosts)} if isinstance(hosts, list) else hosts
+        config['hosts'] = hosts
+
+        return _make_rb_cluster(**config)
+
+    def __str__(self):
+        return 'Redis Blaster Cluster'
+
+
+class _RedisCluster(object):
+    def supports(self, config):
+        return config.get('is_redis_cluster', False)
+
+    def factory(self, **config):
+        # StrictRedisCluster expects a list of { host, port } dicts. Coerce the
+        # configuration into the correct format if necessary.
+        hosts = config.get('hosts')
+        hosts = hosts.values() if isinstance(hosts, dict) else hosts
+
+        # Redis cluster does not wait to attempt to connect, we don't want the
+        # application to fail to boot because of this, raise a KeyError
+        try:
+            return rediscluster.StrictRedisCluster(startup_nodes=hosts, decode_responses=True)
+        except rediscluster.exceptions.RedisClusterException:
+            logger.warning('Failed to connect to Redis Cluster', exc_info=True)
+            raise KeyError('Redis Cluster could not be initalized')
```

Comment:
It's not obvious to me, but will this attempt to reconnect? Or if it fails to connect at startup, is it just dead for the whole life of the process?

Reply:
Right now I'm leaving that up to the caller, so if it makes another attempt later it will try to construct the client again.